### PR TITLE
Redis - added TLS and password, removed core-manager components

### PR DESCRIPTION
**Pull request by tfarinacci**
Was only able to get legacy requirepass to work with Redis.  New ACL command was giving error:
2023-12-13 19:46:20 *** FATAL CONFIG FILE ERROR (Redis 7.2.3) ***
2023-12-13 19:46:20 Reading the configuration file, at line 6
2023-12-13 19:46:20 >>> 'acl setuser codesealer >'password' on allcommands allkeys'
2023-12-13 19:46:20 Bad directive or wrong number of arguments
TLS enabled with self-signed generated cert

### CODE REVIEW

File: charts/codesealer/Chart.yaml [hunk 1]
```diff
@@ -15,10 +15,10 @@ type: application
...
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.1.0
+version: 0.1.1
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
 # It is recommended to use it with quotes.
-appVersion: "0.0.0"
+appVersion: "0.1.0"
```
Comment (psve) [inline charts/codesealer/Chart.yaml]: Not quite sure what to do with this. It is used as a default for image tags, but in it's current form that doesn't make any sense (i.e. we don't use semver). We can set it to our actual release versions, but we don't have images tagged like that yet. I would just leave it for now.

File: tests/test_install.sh
Comment (psve) [inline tests/test_install.sh:1]: Please don't revert the changes I made to this file, they were made for a reason.

File: charts/codesealer/Chart.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.1.1
+version: 0.1.2
```
Comment (psve) [inline charts/codesealer/Chart.yaml]: Do we really need to bump the version for each commit? It's not like we'll actually build a release for each of these.

File: charts/codesealer/templates/config-toml-configmap.yaml [hunk 3]
```diff
@@ -7,12 +7,35 @@ metadata:
...
+    redisUser = {{ .Values.worker.config.bootloader.redisUser | quote }}
+    redisPassword = {{ .Values.worker.config.bootloader.redisPassword | quote }}
+    redisUseTLS = {{ .Values.worker.config.bootloader.redisUseTLS }}
+    redisIgnoreTLS = {{ .Values.worker.config.bootloader.redisIgnoreTLS }}
+    redisConnectTimeout = {{ .Values.worker.config.bootloader.redisConnectTimeout }}
+    redisReadTimeout = {{ .Values.worker.config.bootloader.redisReadTimeout }}
+    redisWriteTimeout = {{ .Values.worker.config.bootloader.redisWriteTimeout }}
+    influxAddress = {{ .Values.influxDB.influxAddress | quote }}
+    influxToken = {{ .Values.influxDB.influxToken | quote }}
+    influxIgnoreTLS = {{ .Values.influxDB.influxIgnoreTLS }}
```
Comment (psve) [inline charts/codesealer/templates/config-toml-configmap.yaml]: What is this doing here if we removed the manager?

File: charts/codesealer/templates/configmap-crs-setup.yaml [hunk 4]
```diff
@@ -695,13 +695,13 @@ data:
...
     #
-    #SecAction \
+    # SecAction \
     # "id:900600,\
     #  phase:1,\
     #  nolog,\
     #  pass,\
     #  t:none,\
-    #  setvar:'tx.high_risk_country_codes='"
+    #  setvar:'tx.high_risk_country_codes={{ .Values.worker.config.endpoint.crs.high_risk_country_codes }}'"
```
Comment (psve) [inline charts/codesealer/templates/configmap-crs-setup.yaml]: Should these be uncommented?

File: tests/test_install.sh [hunk 5]
```diff
@@ -1,192 +1,248 @@
...
+  echo "#   -- Using github.com/bitnami/charts repository"
+  echo "#      Documentation: https://github.com/bitnami/charts/blob/main/bitnami/redis/README.md"
+  echo "########################################################################################"
+  read -p 'Install Redis in single master mode [y/n]: '
+  if [ $REPLY == 'y' ]; then
+    helm install redis-${RELEASE_VER} oci://registry-1.docker.io/bitnamicharts/redis \
+    --namespace ${REDIS_NAMESPACE} --create-namespace \
+    --set auth.enabled=true \
+    --set replica.replicaCount=1 \
+    --wait --timeout=60s
```
Comment (psve) [inline tests/test_install.sh]: The chart readme needs to have section added on redis.

File: charts/codesealer/templates/config-toml-configmap.yaml (hunk 3 above)
Comment (tfarinacci) [reply charts/codesealer/templates/config-toml-configmap.yaml]: I added back due to this documentation under [network] for manager.tls:
      # Paths for TLS certificate and key files that are used if 'manager.tls' is set to true.
      # When using the Management Portal, the certificate and key above will be used as
      # fallback when *no* certificate and key is specified in the Management Portal.

We can remove if TLS is enabled without the manager section.

File: charts/codesealer/templates/configmap-crs-setup.yaml (hunk 4 above)
Comment (tfarinacci) [reply charts/codesealer/templates/configmap-crs-setup.yaml]: Some of the features require us to pull files down to enable.  We need to go through those and determine which we should support.

File: tests/test_install.sh (hunk 5 above)
Comment (tfarinacci) [reply tests/test_install.sh]: Agreed.  Note the documentation adds to the new test script.

File: charts/codesealer/Chart.yaml (hunk 1 above)
Comment (tfarinacci) [reply charts/codesealer/Chart.yaml]: OK.  Defer to you on how to address going forward.

File: tests/test_install.sh
Comment (tfarinacci) [reply tests/test_install.sh:1]: My miss.  I had my git merge flag set incorrectly and overwrote it.

File: charts/codesealer/templates/config-toml-configmap.yaml (hunk 3 above)
Comment (psve) [reply charts/codesealer/templates/config-toml-configmap.yaml]: Looks like a mistake in the docs, it should be `network.tls`. I'll get that fixed.

File: charts/codesealer/templates/configmap-crs-setup.yaml (hunk 4 above)
Comment (psve) [reply charts/codesealer/templates/configmap-crs-setup.yaml]: In that case, we shouldn't put in Helm values that aren't doing anything yet.

Comment (tfarinacci) [reply charts/codesealer/templates/configmap-crs-setup.yaml]: Agreed.  I do think it's something we need to support.  Let's work with Felipe to determine which reputation features make sense to enable.

File: tests/test_install.sh
Comment (psve) [inline tests/test_install.sh:1]: Please run this through [ShellCheck](https://www.shellcheck.net/) - this diff has introduced a bunch of new issues.

File: tests/test_install.sh [hunk 6]
```diff
@@ -1,192 +1,284 @@
...
+  echo "##  Usage: ./test_install.sh [install | uninstall | upgrade]"
+  echo "##"
+  echo "##+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+"
+  exit 1
 fi
 
 clear
 
+# Github Image Registry PAT
+# $CODESEALER_TOKEN=
```
Comment (psve) [inline tests/test_install.sh]: We should explicitly check for this variable being set and warn the caller instead of this.

File: tests/test_install.sh [hunk 7]
```diff
@@ -1,192 +1,284 @@
...
+    echo "#  Waiting for Redis to start"
+    echo "########################################################################################"      
+    helm install redis-${RELEASE_VER} oci://registry-1.docker.io/bitnamicharts/redis \
+    --namespace ${REDIS_NAMESPACE} --create-namespace \
+    --set auth.enabled=true \
+    --set replica.replicaCount=1 \
+    --wait --timeout=60s
+
+    # --set sentinel.enabled=false \
+    # --set auth.sentinel=false \
```
Comment (psve) [inline tests/test_install.sh]: Please don't leave random commented lines lying around.

File: tests/test_install.sh [hunk 8]
```diff
@@ -1,192 +1,284 @@
...
-  --for=condition=ready pod \
-  --selector=app.kubernetes.io/name=codesealer-mutating-webhook \
-  --timeout=90s
+    # --set worker.config.endpoint.hostScheme=https \
+    # --set worker.config.endpoint.hostname=localhost \
+    # --set worker.redis.namespace=codesealer-system \
+    # --set ingress.namespace=codesealer-system \
+    # --set ingress.enabled=true \
+    # --set "ingress.tls[0].hosts[0]=core-manager.local,ingress.tls[0].secretName=ingress-tls" \
+    # --set "ingress.hosts[0].host=core-manager.local,ingress.hosts[0].paths[0].path=/,ingress.hosts[0].paths[0].pathType=Prefix"
```
Comment (psve) [inline tests/test_install.sh]: Same here: If it doesn't serve a purpose, remove it.

File: tests/test_install.sh [hunk 9]
```diff
@@ -1,192 +1,284 @@
...
   read -r -s -p 'Press any key to continue.'
 
-  # Enable Codesealer
-  echo "########################################################################################"
-  echo "  $ kubectl label ns ${INGRESS_NAMESPACE} codesealer.com/webhook=enabled"
-  echo "  $ kubectl patch deployment ${INGRESS_DEPLOYMENT} -n ${INGRESS_NAMESPACE} -p '{"spec": {"template":{"metadata":{"annotations":{"codesealer.com/injection":"enabled"}}}} }'"
-  echo "########################################################################################"
-
   kubectl label ns ${INGRESS_NAMESPACE} codesealer.com/webhook=enabled
+  echo "# "
```
Comment (psve) [inline tests/test_install.sh]: This doesn't seem to belong.

File: tests/test_install.sh [hunk 10]
```diff
@@ -1,192 +1,284 @@
...
   echo "########################################################################################"
   echo "#  Do you wish to uninstall NGINX Ingress Controller?"
   echo "########################################################################################"
   read -r -p 'Uninstall NGINX Ingress Controller [y/n]: '
-
-  if [[ ${REPLY} == 'y' ]]; then
-    helm uninstall ingress-nginx --namespace ${INGRESS_NAMESPACE}
+  if [ $REPLY == 'y' ]; then
+    helm uninstall ingress-nginx-${RELEASE_VER} --namespace ${INGRESS_NAMESPACE}
+    # kubectl delete namespace ${INGRESS_NAMESPACE}
```
Comment (psve) [inline tests/test_install.sh]: Remove this.

File: tests/test_install.sh [hunk 11]
```diff
@@ -1,192 +1,284 @@
...
+    --set worker.config.endpoint.crs.paranoiaLevel=1 \
+    --wait --timeout=90s
 
   echo "########################################################################################"
   echo "#  Upgrade Codesealer"
   echo "########################################################################################"
-
   read -r -s -p 'Press any key to continue.'
   kubectl rollout restart deployments --namespace codesealer-system
-  kubectl rollout status deployments --namespace codesealer-system
```
Comment (psve) [inline tests/test_install.sh:175]: Why was this removed?

File: tests/test_install.sh [hunk 12]
```diff
@@ -1,192 +1,284 @@
...
+
+  echo "########################################################################################"
+  echo "#  Please wait 1 minute..."
+  echo "########################################################################################"
+  sleep 60
```
Comment (psve) [inline tests/test_install.sh]: Kubernetes has built in functionality for waiting on resources, why would you do arbitrary 1 minute waits?

File: tests/test_install.sh [hunk 13]
```diff
@@ -1,192 +1,284 @@
...
 
   echo "########################################################################################"
   echo "#  Upgrade Codesealer Worker Sidecar for NGINX Ingress Controller"
   echo "########################################################################################"
   read -r -s -p 'Press any key to continue.'
-
+  # kubectl rollout restart deployments --namespace ${INGRESS_NAMESPACE}
```
Comment (psve) [inline tests/test_install.sh]: Either remove this or use it. In general I would strongly prefer if we can use a proper `rollout restart`. The only issue I saw with this for Nginx was due to unavailable ports on the host, but that can be fixed with a `controller.updateStrategy.rollingUpdate.maxUnavailable=1` as I have noted in the readme.

File: tests/test_install.sh [hunk 14]
```diff
@@ -1,192 +1,284 @@
 #!/bin/bash
-
 set -ueo pipefail
 
 if [[ "$#" -ne 1 ]]; then
-    echo "Usage: ./test_install.sh [install | uninstall | upgrade]"
-    exit 1
+  echo "##+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+#+"
```
Comment (psve) [inline tests/test_install.sh]: Doesn't matter too much since I only consider this an internal test script, but could we at least be consistent, i.e. "#+#" vs "###".

File: tests/test_install.sh (hunk 7 above)
Comment (tfarinacci) [reply tests/test_install.sh]: I originally intended this script to be a reference for Adam to create the copy paste commands.  It was to illustrate how the parameters could be updated by him.  Do we plan on providing a script to the user or will this be for us to test new releases?  This was here to illustrate if we wanted to turn on Sentinel and test it.

File: tests/test_install.sh (hunk 14 above)
Comment (tfarinacci) [reply tests/test_install.sh]: Agreed.

File: tests/test_install.sh (hunk 13 above)
Comment (tfarinacci) [reply tests/test_install.sh]: Good catch.

File: tests/test_install.sh (hunk 12 above)
Comment (tfarinacci) [reply tests/test_install.sh]: Agreed.

File: tests/test_install.sh (hunk 7 above)
Comment (psve) [reply tests/test_install.sh]: Well, this is a public repository, so yes, we're providing this to our users. That's also why I'm trying to keep it fairly polished. 

If we want to present the possible options to the users, we should do that in the right place, i.e. the readme, either explicitly or by pointing the user to `helm show values`.

**Review by psve — APPROVED** [review]
